### ISSUE TITLE
Training from pre-trained weights (Faster-rcnn/Mask-rcnn)

### ISSUE THREAD

**Issue by sidharths** [tensorflow/models#6076]
### System information
- **What is the top-level directory of the model you are using**:
onject detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04.5
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.12
- **Bazel version (if compiling from source)**:

- **CUDA/cuDNN version**:

- **GPU model and memory**: Tesla V100 16160MB

- **Exact command to reproduce**: Followed steps from https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_locally.md to train locally

I have posted the same question on stackoverflow too with a bounty of 50 points. ([https://stackoverflow.com/questions/54162602/training-from-pre-trained-weights-faster-rcnn-mask-rcnn-object-detection-api](https://stackoverflow.com/questions/54162602/training-from-pre-trained-weights-faster-rcnn-mask-rcnn-object-detection-api))
I am training Mask-RCNN object detection API from pre-trained weights from the [object detection zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md) with the coco 2017 dataset.

I have tried both running locally or using the legacy training script. As soon as the training starts, I am never able to see it resume from the mAP mentioned in the [Zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md) for MaskRCNN or FasterRCNN.

What I want to do is resume training from the mAP of pretrained weights (mentioned 33 for mask_rcnn_resnet101_atrous_coco here at the [Zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md)), _is this possible_?

For instance the very first eval result for masks after 1200 iterations or so during training is given below (using script object_detection/model_main.py from [running locally](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_locally.md) approach).

 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.010
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.010
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.003
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.005
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.005
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.002
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.013

**Comment by rootkitchao** [comment]
In my understanding, you are trying to do fine tuning on the mscoco17 dataset for models in the model zoo.The model in the Model Zoo should be trained using the train and valminusminival splits in the mscoco14 dataset.This is equivalent to the train split in mscoco17.
If you are trying to get a model for the mscoco18 challenge.You should initiate training from Imagenet classification checkpoint.
In addition, 1200 steps don't seem to be enough for fine tuning.Need to make a specific judgment based on the convergence of the loss function.The default hyperparameters in the configuration file in samples/configs/ may also need to be modified. It seems that some of these conditions are set for multiple GPUs.

**Comment by KapoorHitesh** [comment]
I have the same doubt. If we are fine tuning a model on the same data using transfer learning then the mAP should be approximate to the one reported for initial iterations. But the observed mAP is much smaller.

**Comment by KapoorHitesh** [comment]
I saw improvements after setting load_all_detection_checkpoint_vars: true in train_config section.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
